# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/ContentProcess.cpp
# Commit: 6450088b6b73
# Full Hash: 6450088b6b73ffc17c79cd7097b1bfe30d00e207
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2020-09-14 21:29:41
# Regressor Bug: 1627075
# File Overlap Count: 1
# Description:
#   Bug 1656261 - Back out all recent StartupCache work r=RyanVM
#   
#   This backs out all work from bug 1627075 as well as all of its
#   descendents. There were a few conflicts when backing this out but
#   overall it was pretty clean, so I would say it's a fairly mild
# ==============================================================================

diff -r b0fabcd60430 -r 6450088b6b73 dom/ipc/ContentProcess.cpp
--- a/dom/ipc/ContentProcess.cpp	Mon Sep 14 17:32:40 2020 +0000
+++ b/dom/ipc/ContentProcess.cpp	Mon Sep 14 17:00:53 2020 +0000
@@ -84,9 +84,6 @@
   char* prefMapHandle = nullptr;
   char* prefsLen = nullptr;
   char* prefMapSize = nullptr;
-  char* scacheHandle = nullptr;
-  char* scacheSize = nullptr;
-  bool wantsScache = true;
 #if defined(XP_MACOSX) && defined(MOZ_SANDBOX)
   nsCOMPtr<nsIFile> profileDir;
 #endif
@@ -130,11 +127,6 @@
         return false;
       }
       prefMapHandle = aArgv[i];
-    } else if (strcmp(aArgv[i], "-scacheHandle") == 0) {
-      if (++i == aArgc) {
-        return false;
-      }
-      scacheHandle = aArgv[i];
 #endif
 
     } else if (strcmp(aArgv[i], "-prefsLen") == 0) {
@@ -147,13 +139,6 @@
         return false;
       }
       prefMapSize = aArgv[i];
-    } else if (strcmp(aArgv[i], "-scacheSize") == 0) {
-      if (++i == aArgc) {
-        return false;
-      }
-      scacheSize = aArgv[i];
-    } else if (strcmp(aArgv[i], "-noScache") == 0) {
-      wantsScache = false;
     } else if (strcmp(aArgv[i], "-safeMode") == 0) {
       gSafeMode = true;
 
@@ -190,11 +175,6 @@
     return false;
   }
 
-  if (wantsScache) {
-    Unused << mozilla::scache::StartupCache::InitChildSingleton(scacheHandle,
-                                                                scacheSize);
-  }
-
   mContent.Init(IOThreadChild::message_loop(), ParentPid(), *parentBuildID,
                 IOThreadChild::TakeChannel(), *childID, *isForBrowser);
 